lens/lens.go

```diff
@@ -182,18 +182,14 @@ func (l *lens) Next() (bool, error) {
 
 				historyLocation = historyLocation.next.Value()
 			} else {
-				// The pipe head then becomes the schema version migration to the next version
-				// sourcing from any documents at schemaVersionID, or lower schema versions.
-				// This also ensures each document only passes through each migration once,
-				// in order, and through the same state container (in case migrations use state).
-				pipeHead, err = l.lensRegistry.MigrateDown(junctionPipe, historyLocation.schemaVersionID)
+				// Aquire a lens migration from the registery, using the junctionPipe as its source.
+				// The new pipeHead will then be connected as a source to the next migration-stage on
+				// the next loop.
+				pipeHead, err = l.lensRegistry.MigrateDown(junctionPipe, historyLocation.previous.Value().schemaVersionID)
```

Comment:
question: After this line does `historyLocation.previous.Value()` change? if not why not move the declaration of `historyLocation` from below to above here and use here.

Reply:
`Value()` is a very cheap getter, and I probably thought nothing about calling it twice.

I probably bounce around between which I prefer here, as a lot of the time I will prefer the double call over having an extra variable floating around.

Reply:
Is totes fine either way, just thought if there is a var then to use it (or remove it, and call `Value()` everywhere)